### PR TITLE
(CL) MVP route splitting

### PR DESCRIPTION
**Pull request by jonator**
https://app.clickup.com/t/866a13ram

### CODE REVIEW

**Review by arispech — APPROVED** [review]
working fine on my end.

File: packages/pools/src/router/__tests__/routes.spec.ts
Comment (JoseRFelix) [inline packages/pools/src/router/__tests__/routes.spec.ts:16]: I believe we should use `describe` sparingly. Nested code tends to be more difficult to read and maintain. It seems to me that `OptimizedRoutes` and handles do not offer much information, so we could possibly remove these `describe` blocks. Perhaps we could incorporate 'handles' directly within `it`.

File: packages/pools/src/router/__tests__/routes.spec.ts
Comment (JoseRFelix) [inline packages/pools/src/router/__tests__/routes.spec.ts:25]: I also think that `it` statements should be read as full sentences to understand what the test is doing without having to run it within jest.

File: packages/pools/src/router/__tests__/routes.spec.ts
Comment (JoseRFelix) [inline packages/pools/src/router/__tests__/routes.spec.ts:88]: I think we should decide to use either `test` or `it` for consistency. wdyt?

File: packages/pools/src/router/route.ts
Comment (JoseRFelix) [inline packages/pools/src/router/route.ts:16]: Could you expand this comment a little bit more 🙏 , I'm confused as to whether this refers to a method or the interface

File: packages/stores/src/account/msg/amino.ts
Comment (JoseRFelix) [inline packages/stores/src/account/msg/amino.ts:1]: We should keep in mind that we'll have to refactor this file before merging CL or multiple wallets. These amino files are no longer required thanks to the `proto-codecs` package

File: packages/stores/src/account/types.ts
Comment (JoseRFelix) [inline packages/stores/src/account/types.ts:63]: This too

File: packages/web/hooks/ui-config/use-trade-token-in-config.ts
Comment (JoseRFelix) [inline packages/web/hooks/ui-config/use-trade-token-in-config.ts:121]:
```suggestion
      /**
      * Send messages to account
      */
```

File: packages/web/hooks/use-overridable-state.ts
Comment (JoseRFelix) [inline packages/web/hooks/use-overridable-state.ts:1]: I'm somewhat hesitant about renaming this hook to overridable, as it doesn't precisely convey its functionality. In my opinion, the term 'overridable' is quite broad; for instance, `useState` could technically be considered overridable as well. I believe keeping it as `controllable` is a more suitable name, given that the concept of controllable/uncontrollable is well recognized within the React ecosystem. This term shows that the hook's behavior can be switched between uncontrolled and controlled modes as required

**Review by JoseRFelix — CHANGES_REQUESTED** [review]
It's looking good! I've added a few comments for consideration

File: packages/pools/src/router/__tests__/routes.spec.ts
Comment (jonator) [reply packages/pools/src/router/__tests__/routes.spec.ts:16]: Generally I agree in code, but in this scenario IMO it helps keep the types of tests organized given how many there are. It creates a nice tree structure that helps surface the more important tests first, and helps categorize the test ("handles" tests are for testing common error cases/defaults). This tree structure is especially powerful in the jest VSCode extension.

Comment (jonator) [reply packages/pools/src/router/__tests__/routes.spec.ts:16]: Screenshot 2023-05-15 at 12 57 48 PM

File: packages/stores/src/account/msg/amino.ts
Comment (jonator) [reply packages/stores/src/account/msg/amino.ts:1]: Yeah I refactored the swap messages (added one, removed one, refactored 2). I imagine we'll have some nasty conflicts there.

File: packages/stores/src/account/types.ts
Comment (jonator) [reply packages/stores/src/account/types.ts:63]: How does it handle gas?

File: packages/pools/src/router/__tests__/routes.spec.ts
Comment (JoseRFelix) [reply packages/pools/src/router/__tests__/routes.spec.ts:16]: IMO inlining handles like `handles no pools`, `handles invalid tokens`, and `handled not enough liquidity` makes it easier to read for me and also makes me avoid looking consistently up and down. Likewise, if tests fill the list, we would have to constantly scroll up. I guess we could also rely on memory, but I believe inlining it would make it less cumbersome. wdyt?

File: packages/stores/src/account/types.ts
Comment (JoseRFelix) [reply packages/stores/src/account/types.ts:63]: It will be the same object as this one. We'd only have a MessageComposer instead of a type to replace the amino classes. On the other hand, if the dev omits the gas, it's possible to simulate and get an estimated fee

**Review by JoseRFelix — APPROVED** [review]
LGTM! 🚀

**Pull request merged**
